### ISSUE TITLE
Deleting GW pod leads to packet loss in pod2pod ping

### ISSUE THREAD

**Issue by yboaron** [submariner-io/submariner#2228]
I run the following steps:
- Deploy 2 Kind clusters (Kubernetes Version: v1.25.3) 
- Deploy SM 0.14.1 non GN  , IPSec
- Run nettest pod on non GW node on each cluster
- Kubectl exec nettest pod at cluster2
 - ping pod IP of nettest pod at cluster1
- On cluster1 , run script that deletes GW pod every 40 seconds ~16 times

Pod ping summary reports ~20% of packet loss, check [1]   

Also noticed that GW pod on cluster2 failed 7 times during this period, checking GW pod logs [2] on cluster2 seems that Pluto exited due to IKE timeout exceeded 

[1]
608 packets transmitted, 490 received, +8 errors, 19.4079% packet loss, time 621538ms
rtt min/avg/max/mdev = 0.124/0.261/2.199/0.158 ms, pipe 4

[2] 
```
Dec 27 09:58:45.350012: | FOR_EACH_STATE_... in (show_traffic_status() +2401 /programs/pluto/state.c)
Dec 27 09:58:45.350015: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #23
Dec 27 09:58:45.350017: |   found "submariner-cable-cluster-1-172-18-0-5-1-0" #22
Dec 27 09:58:45.350019: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #21
Dec 27 09:58:45.350020: |   found "submariner-cable-cluster-1-172-18-0-5-0-1" #20
Dec 27 09:58:45.350022: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #19
Dec 27 09:58:45.350024: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #18
Dec 27 09:58:45.350026: |   found "submariner-cable-cluster-1-172-18-0-5-1-0" #17
Dec 27 09:58:45.350028: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #16
Dec 27 09:58:45.350030: |   found "submariner-cable-cluster-1-172-18-0-5-0-1" #15
Dec 27 09:58:45.350032: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #14
Dec 27 09:58:45.350033: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #13
Dec 27 09:58:45.350035: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #12
Dec 27 09:58:45.350037: |   found "submariner-cable-cluster-1-172-18-0-5-1-0" #11
Dec 27 09:58:45.350038: |   found "submariner-cable-cluster-1-172-18-0-5-0-1" #9
Dec 27 09:58:45.350040: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #8
Dec 27 09:58:45.350042: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #7
Dec 27 09:58:45.350044: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #6
Dec 27 09:58:45.350046: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #5
Dec 27 09:58:45.350047: |   found "submariner-cable-cluster-1-172-18-0-5-1-0" #4
Dec 27 09:58:45.350049: |   found "submariner-cable-cluster-1-172-18-0-5-1-1" #3
Dec 27 09:58:45.350051: |   found "submariner-cable-cluster-1-172-18-0-5-0-1" #2
Dec 27 09:58:45.350053: |   found "submariner-cable-cluster-1-172-18-0-5-0-0" #1
Dec 27 09:58:45.350055: |   matches: 22
Dec 27 09:58:45.350057: | FOR_EACH_STATE_... in (show_traffic_status() +2401 /programs/pluto/state.c)
--
Dec 27 09:58:45.350254: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350259: | kernel: get_sa_bundle_info esp.da401578@172.18.0.5
Dec 27 09:58:45.350260: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350264: |] #9: "submariner-cable-cluster-1-172-18-0-5-0-1", type=ESP, add_time=1672135025, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350268: | kernel: get_sa_bundle_info esp.6265a2a9@172.18.0.6
Dec 27 09:58:45.350270: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350276: | kernel: get_sa_bundle_info esp.b3981afb@172.18.0.5
Dec 27 09:58:45.350278: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350284: |] #15: "submariner-cable-cluster-1-172-18-0-5-0-1", type=ESP, add_time=1672135067, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350291: | kernel: get_sa_bundle_info esp.aede104a@172.18.0.6
Dec 27 09:58:45.350294: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350300: | kernel: get_sa_bundle_info esp.a91ae2ef@172.18.0.5
Dec 27 09:58:45.350303: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350309: |] #20: "submariner-cable-cluster-1-172-18-0-5-0-1", type=ESP, add_time=1672135108, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350316: | kernel: get_sa_bundle_info esp.14248661@172.18.0.6
Dec 27 09:58:45.350318: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350325: | kernel: get_sa_bundle_info esp.de475357@172.18.0.5
Dec 27 09:58:45.350327: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350333: |] #4: "submariner-cable-cluster-1-172-18-0-5-1-0", type=ESP, add_time=1672134997, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350339: | kernel: get_sa_bundle_info esp.b944e987@172.18.0.6
Dec 27 09:58:45.350342: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350349: | kernel: get_sa_bundle_info esp.9237fe21@172.18.0.5
Dec 27 09:58:45.350351: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350357: |] #11: "submariner-cable-cluster-1-172-18-0-5-1-0", type=ESP, add_time=1672135025, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350364: | kernel: get_sa_bundle_info esp.3d6b136b@172.18.0.6
Dec 27 09:58:45.350367: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350373: | kernel: get_sa_bundle_info esp.53a59989@172.18.0.5
Dec 27 09:58:45.350376: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350387: |] #17: "submariner-cable-cluster-1-172-18-0-5-1-0", type=ESP, add_time=1672135067, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350393: | kernel: get_sa_bundle_info esp.7ed581b3@172.18.0.6
Dec 27 09:58:45.350394: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350401: | kernel: get_sa_bundle_info esp.d0071049@172.18.0.5
Dec 27 09:58:45.350403: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350408: |] #22: "submariner-cable-cluster-1-172-18-0-5-1-0", type=ESP, add_time=1672135108, inBytes=0, outBytes=0, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350415: | kernel: get_sa_bundle_info esp.68e3837@172.18.0.6
Dec 27 09:58:45.350418: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350425: | kernel: get_sa_bundle_info esp.3da2e75@172.18.0.5
Dec 27 09:58:45.350427: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350432: |] #3: "submariner-cable-cluster-1-172-18-0-5-1-1", type=ESP, add_time=1672134997, inBytes=6316, outBytes=7080, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350452: | kernel: get_sa_bundle_info esp.27e78827@172.18.0.6
Dec 27 09:58:45.350455: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350463: | kernel: get_sa_bundle_info esp.2afc89ec@172.18.0.5
Dec 27 09:58:45.350465: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350472: |] #16: "submariner-cable-cluster-1-172-18-0-5-1-1", type=ESP, add_time=1672135067, inBytes=9260, outBytes=10108, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350479: | kernel: get_sa_bundle_info esp.d40f9ae0@172.18.0.6
Dec 27 09:58:45.350482: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350488: | kernel: get_sa_bundle_info esp.18ed76a5@172.18.0.5
Dec 27 09:58:45.350491: | xfrm: sendrecv_xfrm_msg() sending 18
Dec 27 09:58:45.350498: |] #21: "submariner-cable-cluster-1-172-18-0-5-1-1", type=ESP, add_time=1672135108, inBytes=4356, outBytes=4356, maxBytes=2^63B, id='172.18.0.5'
Dec 27 09:58:45.350503: | whack: stop: trafficstatus (fd@0x5652bc6f0c58)
Dec 27 09:58:45.350512: | freeref fd@0x5652bc6f0c58 (whack_handle_cb() +1038 /programs/pluto/rcv_whack.c)
Dec 27 09:58:47.178400: | timer_event_cb: processing EVENT_RETRANSMIT-event@0x5652bc6c17e8 for IKE SA #8 in state ESTABLISHED_IKE_SA
Dec 27 09:58:47.178411: | #8 deleting EVENT_RETRANSMIT
Dec 27 09:58:47.178415: | IKEv2 retransmit event
Dec 27 09:58:47.178418: | handling event EVENT_RETRANSMIT for 172.18.0.5 "submariner-cable-cluster-1-172-18-0-5-1-1" #8 attempt 1 of 0
Dec 27 09:58:47.178420: | and parent for 172.18.0.5 "submariner-cable-cluster-1-172-18-0-5-1-1" #8 keying attempt 0 of 0; retransmit 8
Dec 27 09:58:47.178422: | #8 STATE_V2_ESTABLISHED_IKE_SA: retransmits: current time 1739800.228105
Dec 27 09:58:47.178424: | #8 STATE_V2_ESTABLISHED_IKE_SA: retransmits: retransmit count 7 exceeds limit? NO
Dec 27 09:58:47.178425: | #8 STATE_V2_ESTABLISHED_IKE_SA: retransmits: deltatime 64 exceeds limit? YES
Dec 27 09:58:47.178427: | #8 STATE_V2_ESTABLISHED_IKE_SA: retransmits: monotime 64.0137 exceeds limit? YES
Dec 27 09:58:47.178429: "submariner-cable-cluster-1-172-18-0-5-1-1" #8: STATE_V2_ESTABLISHED_IKE_SA: 60 second timeout exceeded after 7 retransmits.  No response (or no acceptable response) to our IKEv2 message
Dec 27 09:58:47.178435: ABORT: ASSERTION FAILED: switch (ike->sa.st_connection->config->dpd.action) case 0 (0x0) unexpected (retransmit_timeout_action() +100 /programs/pluto/ikev2_retry.c)
2022-12-27T09:58:47.462Z FTL ...1/pkg/log/logger.go:67 libreswan            Pluto exited: signal: aborted (core dumped)
```

**Comment by yboaron** [comment]
I tested a similar test this time with VxLAN as a cable driver,  
while the remote GW pod didn't fail,  ~50% of pod2pod ping traffic was lost [1].



[1] 

--- 10.130.0.6 ping statistics ---
1053 packets transmitted, 505 received, 52.0418% packet loss, time 1077192ms
rtt min/avg/max/mdev = 0.118/0.225/0.789/0.109 ms
/

**Comment by yboaron** [comment]
I reproduced the issue in Kind and checked the logs of restarted GW pod, and it seems that [main loop controllers cleanup code ](https://github.com/submariner-io/submariner/blob/devel/main.go#L205-#L215) is called  which may explain ping failure.

The 'kubectl delete pod' sends SIGTERM and SIGKILL signals to the pod and in current GW pod [code](https://github.com/submariner-io/submariner/blob/devel/main.go#L205) it leads to DP cleanup

**Comment by yboaron** [comment]
Regarding the other use case, restarting of active GW pod several times due to infra failures of K8s for example.

- For GW pod we use the default restartPolicy (==Always), meaning K8S will try to restart the pod and use a back-off timer (10,20,40, etc seconds) up to a maximum of 5 minutes.

Assuming we have two clusters and the GW pod in Cluster1 is restarted several times because of failures in K8S infrastructure

A. K8S will restart the GW pod in Cluster1 using back-off timer values

B. No packet loss was observed during the first restarts of the GW pod in cluster1

C. When the back-off timer value is large, the GW pod in Cluster2 will crash due to
  Pluto failure (check [1]), this failure clears the IPSec DP configuration on cluster 2, so we have packet loss from that point

D. When GW pod in cluster2 restarts it gets stuck in NAT discovery phase, because cluster1's GW pod is down, so no IPSec DP configuration at this phase (check [2] )

E. After NAT discovery timeout expired , IPSec DP configured on cluster2 , but still no IPSec active connection  (check [3] )  

E. IPSec traffic recovered only after GW pod in cluster1 started running, we have packet loss of 5 minutes.

The bottom line, several restarts of GW pod lead to packet loss for a period of ~5 minutes.

cc @sridhargaddam 

[1] 
connections obtained from whack: map[], map[]
2023-01-10T16:51:11.013Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:51:11.013Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:51:11.013Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:51:11.013Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:51:11.013Z DBG ..reswan/libreswan.go:266 libreswan            Connection "submariner-cable-cluster1-172-18-0-5" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:51:15.830Z DBG ..ry/request_handle.go:52 NAT                  Received request from 172.18.0.5:4490 - REQUEST_NUMBER: 0xdbd0756b21e9db00, SENDER: "submariner-cable-cluster1-172-18-0-5", RECEIVER: "submariner-cable-cluster2-172-18-0-4"
2023-01-10T16:51:15.830Z DBG ..y/request_handle.go:117 NAT                  Sending response to 172.18.0.5:4490 - REQUEST_NUMBER: 0xdbd0756b21e9db00, RESPONSE: OK, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5"
2023-01-10T16:53:58.169Z FTL ..m1/pkg/log/logger.go:67 libreswan            Pluto exited: signal: aborted (core dumped)

[2] 

2023-01-10T17:21:21.668Z INF ../datastoresyncer.go:297 DSSyncer             Creating local submariner Cluster: types.SubmarinerCluster{ID:"cluster2", Spec:v1.ClusterSpec{ClusterID:"cluster2", ColorCodes:[]string{"blue"}, ServiceCIDR:[]string{"100.67.0.0/16"}, ClusterCIDR:[]string{"10.131.0.0/16"}, GlobalCIDR:[]string{}}} 
2023-01-10T17:21:21.674Z INF ../datastoresyncer.go:310 DSSyncer             Creating local submariner Endpoint: types.SubmarinerEndpoint{Spec:v1.EndpointSpec{ClusterID:"cluster2", CableName:"submariner-cable-cluster2-172-18-0-4", HealthCheckIP:"10.131.1.1", Hostname:"cluster2-worker", Subnets:[]string{"100.67.0.0/16", "10.131.0.0/16"}, PrivateIP:"172.18.0.4", PublicIP:"79.176.53.20", NATEnabled:false, Backend:"libreswan", BackendConfig:map[string]string{"natt-discovery-port":"4490", "preferred-server":"false", "udp-port":"4500"}}} 
2023-01-10T17:21:21.679Z INF ../datastoresyncer.go:104 DSSyncer             Datastore syncer started
2023-01-10T17:21:22.405Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf7f, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:22.405Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf80, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:24.405Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf81, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:24.406Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf82, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:26.407Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf83, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:26.407Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf84, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:28.409Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf85, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:28.409Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf86, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:30.410Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf87, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:30.411Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf88, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:32.411Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf89, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:32.411Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8a, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:34.413Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8b, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:34.414Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8c, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:36.415Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8d, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:36.415Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8e, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:38.416Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf8f, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:38.416Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf90, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:40.418Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf91, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:40.418Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf92, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:42.418Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf93, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:42.418Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf94, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:44.419Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf95, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:44.419Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf96, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:46.421Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf97, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:46.421Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf98, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:48.422Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf99, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:48.422Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9a, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:50.424Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9b, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:50.424Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9c, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:52.426Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9d, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:52.426Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9e, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:54.427Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bf9f, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:54.428Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa0, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:56.429Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa1, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:56.430Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa2, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:21:58.431Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa3, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:21:58.432Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa4, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:00.432Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa5, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:00.433Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa6, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:02.434Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa7, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:02.434Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa8, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:04.436Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfa9, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:04.436Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfaa, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:06.437Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfab, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:06.437Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfac, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:08.439Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfad, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:08.439Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfae, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:10.441Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfaf, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:10.441Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb0, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:12.442Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb1, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:12.443Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb2, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:14.444Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb3, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:14.444Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb4, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:16.445Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb5, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:16.445Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb6, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:18.446Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb7, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:18.446Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb8, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:20.447Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfb9, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 172.18.0.5:4490
2023-01-10T17:22:20.447Z DBG ..ery/request_send.go:115 NAT                  Sending request - REQUEST_NUMBER: 0x5537e663cc13bfba, SENDER: "submariner-cable-cluster2-172-18-0-4", RECEIVER: "submariner-cable-cluster1-172-18-0-5", USING_SRC: 172.18.0.4:4490, USING_DST: 79.176.53.20:4490
2023-01-10T17:22:22.449Z WRN ..ery/natdiscovery.go:184 NAT                  NAT discovery for endpoint "submariner-cable-cluster1-172-18-0-5" has timed out

[3] 
connections obtained from whack: map[], map[]
2023-01-10T16:55:51.183Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:51.183Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:51.183Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:51.183Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:51.183Z DBG ..reswan/libreswan.go:266 libreswan            Connection "submariner-cable-cluster1-172-18-0-5" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:56.200Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:56.200Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-0-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:56.200Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-0" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:56.200Z DBG ..reswan/libreswan.go:252 libreswan            Connection "submariner-cable-cluster1-172-18-0-5-1-1" not found in active connections obtained from whack: map[], map[]
2023-01-10T16:55:56.200Z DBG ..reswan/libreswan.go:266 libreswan            Connection "submariner-cable-cluster1-172-18-0-5" not found in active connections obtained from whack: map[], map[]

**Comment by sridhargaddam** [comment]
FWIW, a separate issue is reported to track the issue seen with the above observation - https://github.com/submariner-io/submariner/issues/2243#issuecomment-1378655581

**Comment by sridhargaddam** [comment]
@yboaron I see that the [following PR](https://github.com/submariner-io/submariner/commit/cd3e40c566ed57353dd5287766162259bcf3bc06) added the cableEngine.cleanup when the Gateway pod is terminated. From what I remember, prior to this change, we were not seeing the datapath loss when the gateway pod was restarted. Its worth checking if the following description mentioned in the PR is still relevant with the latest OVNK deployments if we revert the change.

CC: @aswinsuryan 

```
With the OVN-K CNI we were seeing and ungraceful
termination of the GW Pod in GW migration tests,
this patch ensures we cleanup the cableengine
on node resources before exiting.
```

**Comment by yboaron** [comment]
@aswinsuryan Any idea if we need [the PR](https://github.com/submariner-io/submariner/commit/cd3e40c566ed57353dd5287766162259bcf3bc06) @sridhargaddam  mentioned for OVN-K  (and how it affects/related to OVN-K) ?  IIUC, it deletes all cable-engine DP configuration (for example delete all Xfrm rules for IPSec) when GW is deleted.

If it's something critical for OVN-K I think we can keep it and close this PR with a comment.

**Comment by dfarrell07** [comment]
Talking with @yboaron, I think we can close this. We seem to understand the root cause, what to avoid.

**Comment by ishankgoel** [comment]
@dfarrell07 
I am facing similar issue on openshift cluster with submariner version 0.22.0 where packets are being dropped due to stale IPSec XFRM configuration left by previous active gateway pod. I couldn't identify a fix for this from previous comments.
